### PR TITLE
Migrate to openldap 2.5+

### PR DESCRIPTION
**Pull request by ellert**
# Summary

Issue #41 describes how updating to openldap 2.5 or later breaks bdii due to the removal of the hdb slapd backend used by bdii. The issue is now more urgent, since RedHat Enterprise Linux 9, and derivatives such as CentOS Stream 9 and Rocky Linux 9, uses openldap 2.6.

This PR proposes a migration to the mdb slapd backend. The changes were first proposed by andreas.haupt@desy.de in a bugzilla ticket for the EPEL 9 version of bdii in the RedHat bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=2150344

---
**Related issue: #41**

### CODE REVIEW

File: debian/rules [hunk 1]
```diff
@@ -6,7 +6,7 @@
 override_dh_auto_install:
 	$(MAKE) prefix=debian/bdii install
 	slapd_modulepath="modulepath      /usr/lib/ldap" ; \
-	slapd_moduleload="moduleload      back_hdb" ; \
+	slapd_moduleload="moduleload      back_mdb" ; \
```
Comment (gwarf) [inline debian/rules:9]: Do you know starting with which debian version we should use mdb?

**Comment by gwarf** [comment]
Thanks a lot @ellert!
We have seen that in https://src.fedoraproject.org/rpms/bdii/c/68201bd332defa7cd8fba764c5b2ade85cdd08fe?branch=rawhide you are using a patch to do this only for >= C9, I guess we may want to do the same.

So can we borrow your patch? or maybe you want to add it yourself to this PR?

We have discussed this with @andrea-manzi and we plan to:
- build a version for CentOS 7 (and  CentOS 8 Stream in case one wants it) with python3 and hdb support
- build a version for CentOS 9  Stream with python3 and mdb
- To be clarified what we do for debian, as debian bullesye is still using [slapd 2.4](https://packages.debian.org/bullseye/slapd), so it may be safer to also use mdb conditionnaly on debian, what do you think? For debian we could also add building to the GitHub actions pipeline. How do you want to proceed with this?

**Comment by gwarf** [comment]
You should also have received an invitation to join the EGI Federation org, if you accept it would grant you write rights to this repo and to a bunch of other BDII repos, I guess it makes sense.

File: debian/rules (hunk 1 above)
Comment (gwarf) [reply debian/rules:9]: OK, looks like 2.5 is present starting with [bookworm](https://packages.debian.org/bookworm/slapd), while [bullseye](https://packages.debian.org/bullseye/slapd) is still 2.4.

**Comment by ellert** [comment]
> So can we borrow your patch? or maybe you want to add it yourself to this PR?

The patch in the Fedora/EPEL build is generated from the git commit in the PR:

git format-patch a3312f93c372f9a8dd420fb991d04383531faae6~1..a3312f93c372f9a8dd420fb991d04383531faae6

**Review by enolfc — APPROVED** [review]
LGTM!

**Pull request merged**